kubeflow/components/train-spl.yaml

```diff
@@ -0,0 +1,145 @@
+name: Train
+description: |
+  Training
+inputs:
+  - {name: image, description: ''}
+  - {name: s3_bucket, description: ''}
+  - {name: owner, description: ''}
+  - {name: task_name, description: ''}
+  - {name: project, description: ''}
+  - {name: experiment, description: ''}
+  - {name: model, description: ''}
+  - {name: load_from, description: ''}
+  - {name: eval_set, description: ''}
+  - {name: s3_datadir, description: ''}
+  - {name: s3_database_dir, description: ''}
+  - {name: dataset_subfolder, description: ''}
+  - {name: train_iterations, description: ''}
+  - {name: skip_tensorboard, description: ''}
+  - {name: train_languages, description: ''}
+  - {name: eval_languages, description: ''}
+  - {name: dlg_side, description: ''}
+  - {name: s3_bootleg_prepped_data, description: ''}
+  - {name: bootleg_model, description: ''}
+  - {name: additional_args, description: ''}
+outputs:
+  - {name: s3_model_dir}
+implementation:
+  container:
+    image: '{{inputs.parameters.image}}'
+    command:
+    - /bin/bash
+    - -ex
+    - -c
+    - |
+      . /opt/genie-toolkit/lib.sh
+
+      parse_args "$0" "image s3_bucket owner task_name project experiment model load_from s3_datadir s3_database_dir dataset_subfolder s3_model_dir skip_tensorboard train_languages eval_languages eval_set dlg_side s3_bootleg_prepped_data bootleg_model" "$@"
+      shift $n 
+      cd $HOME
+
+      /opt/genie-toolkit/sync-repos.sh
+
+      IFS='+' ; read -ra TLS <<<"$train_languages"; IFS=' '
+      echo "${TLS[@]}"
+      IFS='+'; read -ra ELS <<<"$eval_languages"; IFS=' '
+      echo "${ELS[@]}"
+
+      modeldir="$HOME/models/$model"
+      mkdir -p "$modeldir"
+
+      if test "${s3_database_dir}" != None ; then
+        aws s3 sync --no-progress ${s3_database_dir} ./database/ --exclude '*' --include '*wikidataqid_to_bootlegtypeid.json' --include '*entityQID_to_wikidataTypeQID.json' --include "*${bootleg_model}/*.json"
+      fi
```

Comment:
How is this component different from regular training? Can you accomplish the same using flags instead?

Reply:
It started differently but I can merge them now.